src/main/java/org/opentripplanner/transit/model/timetable/RealTimeTripTimes.java

```diff
@@ -0,0 +1,584 @@
+package org.opentripplanner.transit.model.timetable;
+
+import static org.opentripplanner.transit.model.timetable.TimetableValidationError.ErrorCode.NEGATIVE_DWELL_TIME;
+import static org.opentripplanner.transit.model.timetable.TimetableValidationError.ErrorCode.NEGATIVE_HOP_TIME;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.OptionalInt;
+import java.util.function.IntUnaryOperator;
+import javax.annotation.Nullable;
+import org.opentripplanner.framework.i18n.I18NString;
+import org.opentripplanner.model.BookingInfo;
+import org.opentripplanner.transit.model.basic.Accessibility;
+import org.opentripplanner.transit.model.framework.DataValidationException;
+
+/**
+ * A TripTimes represents the arrival and departure times for a single trip in an Timetable. It is
+ * carried along by States when routing to ensure that they have a consistent, fast view of the trip
+ * when realtime updates have been applied. All times are expressed as seconds since midnight (as in
+ * GTFS).
+ */
+public final class RealTimeTripTimes implements TripTimes {
```

Comment:
We are not always consistent with this but I find `RealtimeTripTimes` (first t in lower case) more correct.

This maps to my mental model of saying, for example, "realtime updater" rather than "real time updater" or "real-time updater" even though all of them would be grammatically correct.

Reply:
I would be in favour of also renaming to `RealtimeState` and `RealtimeStopState`.

Reply:
This was discussed in a dev meeting. We decided to go with "RealTime" here, since "realtime" has not made it into the dictionaries jet. It probably will be included, but the form "real time"/"real-time" is more common.